### PR TITLE
enhancement(jiva describe): Add missing replica related details to Jiva Volume and PVC describe

### LINKED ISSUE #82: Enhance: Missing details about Jiva Replicas
While describing a Jiva Volume, the following is displayed. 

```
kubectl openebs describe volume --openebs-namespace=openebs pvc-1b21ac95-fd9f-466f-a39b-c1e1ab6e6cb5
pvc-1b21ac95-fd9f-466f-a39b-c1e1ab6e6cb5 Details :
-----------------
NAME            : pvc-1b21ac95-fd9f-466f-a39b-c1e1ab6e6cb5
ACCESS MODE     : ReadWriteOnce 
CSI DRIVER      : jiva.csi.openebs.io
STORAGE CLASS   : jiva-sc-mongo
VOLUME PHASE    : Bound
VERSION         : 2.11.0
JVP             : jiva-policy-mongo
SIZE            : 4.7 GiB
STATUS          : RW
REPLICA COUNT   : 1
Portal Details :
------------------
IQN              :  iqn.2016-09.com.openebs.jiva:pvc-1b21ac95-fd9f-466f-a39b-c1e1ab6e6cb5
VOLUME NAME      :  pvc-1b21ac95-fd9f-466f-a39b-c1e1ab6e6cb5
TARGET NODE NAME :  gke-kmova-helm-default-pool-30f2c6c6-qf2w
PORTAL           :  10.3.252.131:3260
Replica Details :
-----------------
NAME                                                          STATUS   VOLUME                                     CAPACITY   STORAGECLASS       AGE
openebs-pvc-1b21ac95-fd9f-466f-a39b-c1e1ab6e6cb5-jiva-rep-0   Bound    pvc-21b23326-6ec0-463e-9926-d71e467221f4   5.0 GiB    openebs-hostpath   35m
```

This is missing some critical information about a replica, that is available in the JivaVoume CR like:
- Replica Pod Name
- Replica Status (RW or WO or NA )
- Replica Node Name ( that is available within the PVC. 

In fact what is shown as Replica Details in the above output are actually "Replica Pod PVCs"

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
### What does this PR do?
- Rename `Replica Details` ---> `Replica Volume Data Details`.
-  Adds one more column MODE which will denote the (RW or WO or NA ), under `Controller and Replica Pod Details` section.
- Add the `Controller and Replica Pod Details` and `Replica Volume Data Details` sections in Jiva Volume describe as well.
- Closes #82 

### Usage and Output
- Jiva PVC Describe
```
❯ kubectl openebs describe pvc jiva-csi-pvc

jiva-csi-pvc Details  :
-------------------
NAME               : jiva-csi-pvc
NAMESPACE          : default
CAS TYPE           : jiva
BOUND VOLUME       : pvc-d6349231-a863-4d01-962e-24a4b372050d
ATTACHED TO NODE   : node2-virtual-machine
JIVA VOLUME POLICY : jivavolumepolicy
STORAGE CLASS      : openebs-jiva-csi-sc
SIZE               : 4Gi
JV STATUS          : RW
PV STATUS          : Bound

Portal Details :
------------------
IQN              :  iqn.2016-09.com.openebs.jiva:pvc-d6349231-a863-4d01-962e-24a4b372050d
VOLUME NAME      :  pvc-d6349231-a863-4d01-962e-24a4b372050d
TARGET NODE NAME :  node2-virtual-machine
PORTAL           :  10.106.182.76:3260

Controller and Replica Pod Details :
-----------------------------------
NAMESPACE   NAME                                                              MODE   NODE                    STATUS    IP             READY   AGE
jiva        pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-ctrl-575cc6cgj272   RW     node2-virtual-machine   Running   192.168.2.33   1/1     22h28m
jiva        pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-0               RW     node2-virtual-machine   Running   192.168.2.35   1/1     22h28m
jiva        pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-1               RW     node1-virtual-machine   Running   192.168.1.21   1/1     22h28m

Replica Data Volume Details :
-----------------------------
NAME                                                          STATUS   VOLUME                                     CAPACITY   STORAGECLASS       AGE
openebs-pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-0   Bound    pvc-19af6af5-34e1-4ab4-8401-07269c4084fa   4.0GiB     openebs-hostpath   22h28m
openebs-pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-1   Bound    pvc-50a75a4d-a809-45e0-8b48-167e8e5a2c17   4.0GiB     openebs-hostpath   22h28m

```
- Jiva Volume Describe

```
❯ kubectl openebs describe vol pvc-d6349231-a863-4d01-962e-24a4b372050d

pvc-d6349231-a863-4d01-962e-24a4b372050d Details :
-----------------
NAME            : pvc-d6349231-a863-4d01-962e-24a4b372050d
ACCESS MODE     : ReadWriteOnce 
CSI DRIVER      : jiva.csi.openebs.io
STORAGE CLASS   : openebs-jiva-csi-sc
VOLUME PHASE    : Bound
VERSION         : 2.11.0
JVP             : jivavolumepolicy
SIZE            : 4.0GiB
STATUS          : RW
REPLICA COUNT	: 2


Portal Details :
------------------
IQN              :  iqn.2016-09.com.openebs.jiva:pvc-d6349231-a863-4d01-962e-24a4b372050d
VOLUME NAME      :  pvc-d6349231-a863-4d01-962e-24a4b372050d
TARGET NODE NAME :  node2-virtual-machine
PORTAL           :  10.106.182.76:3260

Controller and Replica Pod Details :
-----------------------------------
NAMESPACE   NAME                                                              MODE   NODE                    STATUS    IP             READY   AGE
jiva        pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-ctrl-575cc6cgj272   RW     node2-virtual-machine   Running   192.168.2.33   1/1     22h28m
jiva        pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-0               RW     node2-virtual-machine   Running   192.168.2.35   1/1     22h28m
jiva        pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-1               RW     node1-virtual-machine   Running   192.168.1.21   1/1     22h28m

Replica Data Volume Details :
-----------------------------
NAME                                                          STATUS   VOLUME                                     CAPACITY   STORAGECLASS       AGE
openebs-pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-0   Bound    pvc-19af6af5-34e1-4ab4-8401-07269c4084fa   4.0GiB     openebs-hostpath   22h28m
openebs-pvc-d6349231-a863-4d01-962e-24a4b372050d-jiva-rep-1   Bound    pvc-50a75a4d-a809-45e0-8b48-167e8e5a2c17   4.0GiB     openebs-hostpath   22h28m

```

### CODE REVIEW

**Review by shubham14bajpai — APPROVED** [review]
LGTM

File: pkg/persistentvolumeclaim/jiva.go [hunk 1]
```diff
@@ -50,7 +51,8 @@ func DescribeJivaVolumeClaim(c *client.K8sClient, pvc *corev1.PersistentVolumeCl
 	// 1. Get the JivaVolume Corresponding to the pvc name
 	jv, err := c.GetJV(pvc.Spec.VolumeName)
 	if err != nil {
-		fmt.Printf("failed to get JivaVolume for %s\n", pvc.Spec.VolumeName)
+		fmt.Printf("failed to get JivaVolume for %s", pvc.Spec.VolumeName)
```
Comment (vharsh) [inline pkg/persistentvolumeclaim/jiva.go]: Please print error messages to stderr

File: pkg/persistentvolumeclaim/jiva.go [hunk 2]
```diff
@@ -68,34 +70,66 @@ func DescribeJivaVolumeClaim(c *client.K8sClient, pvc *corev1.PersistentVolumeCl
 	}
 	if vol != nil {
 		jivaPvcInfo.PVStatus = vol.Status.Phase
+		// 3. Print the Jiva Volume Claim information
+		_ = util.PrintByTemplate("jivaPvcInfo", jivaPvcInfoTemplate, jivaPvcInfo)
+	} else {
+		_ = util.PrintByTemplate("jivaPvcInfo", jivaPvcInfoTemplate, jivaPvcInfo)
+		fmt.Printf("PersistentVolume %s, doesnot exist", pvc.Spec.VolumeName)
```
Comment (vharsh) [inline pkg/persistentvolumeclaim/jiva.go]: stderr

**Review by vharsh — APPROVED** [review]

File: pkg/persistentvolumeclaim/jiva.go [hunk 3]
```diff
@@ -68,34 +70,66 @@ func DescribeJivaVolumeClaim(c *client.K8sClient, pvc *corev1.PersistentVolumeCl
...
 	} else {
-		fmt.Printf("Controller and Replica Pod Details :\n-----------------------------------\nNo Controller and Replica pod exists for the JivaVolume\n")
+		fmt.Printf("Controller and Replica Pod Details :")
+		fmt.Println("-----------------------------------")
+		fmt.Println("No Controller and Replica pod exists for the JivaVolume")
 	}
 	// 6. Fetch the replica PVCs and create rows for cli-runtime
 	var rows []metav1.TableRow
-	pvcList, err := c.GetPVCs(jv.Namespace, nil, "openebs.io/component=jiva-replica,openebs.io/persistent-volume="+jv.Name)
+	pvcList, err := c.GetPVCs(c.Ns, nil, "openebs.io/component=jiva-replica,openebs.io/persistent-volume="+vol.Name)
```
Comment (vharsh) [inline pkg/persistentvolumeclaim/jiva.go:131]: The replica PVCs aren't supposed to be present in the JIVA_NAMESPACE?, i.e. was this change to `c.Ns` intended?

Comment (Abhinandan-Purkait) [reply pkg/persistentvolumeclaim/jiva.go:131]: - The above caused a NPE when a JV goes missing. The JIVA_NAMESPACE gets assigned to the client before the method is called. Ref:-
https://github.com/openebs/openebsctl/blob/a5be7503ba4a326545073538980fdcc8fe24e51d/pkg/persistentvolumeclaim/persistentvolumeclaim.go#L52

**Review by vharsh — APPROVED** [review]
lgtm

**Pull request merged**